### ISSUE TITLE
`No such file or directory (os error 2)` when downloading modpack

### ISSUE THREAD

**Issue by insanity54** [gorilla-devs/ferium#402]
### Description

I've come across this error a few times when downloading mods from modrinth. It affects some modpacks and not others. I haven't found a way around it so when it happens I just give up on that specific modpack and move onto a different one.

### Steps to reproduce

```
    ~/Documents/etmc-web    main ?9  mkdir ~/Documents/etmc3069                                      ✔  ▼ 
    ~/Documents/etmc-web    main ?9  cd ~/Documents/etmc3069                                         ✔  ▼ 
direnv: unloading
    ~/Documents/etmc3069  ls                                                                                ✔ 
    ~/Documents/etmc3069  ls\                                                                               ✔ 
> 
    ~/Documents/etmc3069  ferium profile create                                                         INT ✘ 
The default mods directory is /home/chris/.minecraft/mods
✔ Would you like to specify a custom mods directory? · yes
? Pick an output directory (/home/chris/.minecraft/mods) › ./mods
    ~/Documents/etmc3069  mkdir mods                                                             INT ✘  8s  
    ~/Documents/etmc3069  ferium profile create                                                             ✔ 
The default mods directory is /home/chris/.minecraft/mods
✔ Would you like to specify a custom mods directory? · yes
✔ Output Directory · /home/chris/Documents/etmc3069/mods
✔ What should this profile be called? · etmc3069
✔ Which version of Minecraft do you play? · 1.20.1
✔ Which mod loader do you use? · Fabric
After adding your mods, remember to run `ferium upgrade` to download them!
    ~/Documents/etmc3069  ferium modpack add f3FXsUg6                                               ✔  16s  
Checking modpack... ✓ (Genuine)
Where should the modpack be installed to?
? Pick an output directory (/home/chris/.minecraft) › 
    ~/Documents/etmc3069  ferium profile switch                                                  INT ✘  5s  
✔ Select which profile to switch to · Fabric 1.20.1  etmc3069 (0 mods)
    ~/Documents/etmc3069  ferium modpack add f3FXsUg6                                                       ✔ 
Checking modpack... ✓ (Genuine)
Where should the modpack be installed to?
✔ Output Directory · /home/chris/Documents/etmc3069/mods
✔ Should overrides be installed? · yes
WARNING: Configs in your output directory may be overwritten by modpack overrides
    ~/Documents/etmc3069  ferium modpack upgrade                                                     ✔  9s  
Downloading Modpack

Downloading 151 Mod Files

✓ Downloaded 99.1 KB DiagonalFences-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 47.2 KB antighost-1.20-fabric0.83.0-1.1.5.jar
✓ Downloaded 371 KB  YungsBetterWitchHuts-1.20-Fabric-3.0.1.jar
✓ Downloaded 364 KB  balm-fabric-1.20-7.0.7.jar
✓ Downloaded 57.3 KB Prism-1.20.1-fabric-1.0.5.jar
✓ Downloaded 442 KB  3dskinlayers-fabric-1.5.4-mc1.20.1.jar
✓ Downloaded 337 KB  TravelersTitles-1.20-Fabric-4.0.1.jar
✓ Downloaded 443 KB  carryon-fabric-1.20.1-2.1.1.3.jar
✓ Downloaded 295 KB  camerautils-1.20.1-1.0.5.jar
✓ Downloaded 125 KB  InvMove-1.20-0.8.4-Fabric.jar
✓ Downloaded 80.2 KB Searchables-fabric-1.20.1-1.0.1.jar
✓ Downloaded 508 KB  YungsBridges-1.20-Fabric-4.0.1.jar
✓ Downloaded 129 KB  EasyAnvils-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 576 KB  architectury-9.1.12-fabric.jar
✓ Downloaded 57.7 KB EnchantmentDescriptions-Fabric-1.20.1-17.0.4.jar
✓ Downloaded 77.3 KB MouseTweaks-fabric-mc1.20-2.25.jar
✓ Downloaded 317 KB  TerraBlender-fabric-1.20.1-3.0.0.169.jar
✓ Downloaded 124 KB  AxesAreWeapons-1.7.1-fabric-1.19.3.jar
✓ Downloaded 541 KB  YungsExtras-1.20-Fabric-4.0.1.jar
✓ Downloaded 734 KB  PuzzlesLib-v8.0.24-1.20.1-Fabric.jar
✓ Downloaded 618 KB  collective-1.20.1-6.65.jar
✓ Downloaded 172 KB  MixinConflictHelper-1.2.0.jar
✓ Downloaded 338 KB  Debugify-1.20.1+2.0.jar
✓ Downloaded 97.4 KB DiagonalWindows-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 314 KB  Bookshelf-Fabric-1.20.1-20.0.3.jar
✓ Downloaded 23.1 KB clean_tooltips-1.0-fabric-1.20.1.jar
✓ Downloaded 1.44 MB YungsBetterJungleTemples-1.20-Fabric-2.0.1.jar
✓ Downloaded 48.1 KB LeaveMyBarsAlone-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 1.13 MB CreativeCore_FABRIC_v2.10.28_mc1.20.1.jar
✓ Downloaded 222 KB  cardinal-components-api-5.2.2.jar
✓ Downloaded 49.5 KB chat_heads-0.10.20-fabric-1.20.jar
✓ Downloaded 90.6 KB Amplified_Nether_1.20.1_v1.2.2.jar
✓ Downloaded 1.16 MB cloth-config-11.1.106-fabric.jar
✓ Downloaded 1.06 MB appleskin-fabric-mc1.20-2.5.0.jar
✓ Downloaded 169 KB  SnowRealMagic-1.20.1-fabric-9.0.1.jar
✓ Downloaded 69.8 KB animatica-0.6+1.20.jar
✓ Downloaded 368 KB  Nullscape_1.20.1_v1.2.2.jar
✓ Downloaded 80.8 KB UniversalBoneMeal-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 1.17 MB YungsApi-1.20-Fabric-4.0.1.jar
✓ Downloaded 711 KB  comforts-fabric-6.3.3+1.20.1.jar
✓ Downloaded 123 KB  Controlling-fabric-1.20.1-12.0.2.jar
✓ Downloaded 105 KB  EasyMagic-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 114 KB  CullLessLeaves-1.3.0.jar
✓ Downloaded 123 KB  Iceberg-1.20.1-fabric-1.1.15.jar
✓ Downloaded 514 KB  YungsBetterMineshafts-1.20-Fabric-4.0.1.jar
✓ Downloaded 1.77 MB BetterF3-7.0.1-Fabric-1.20.1.jar
✓ Downloaded 31.4 KB LetSleepingDogsLie-1.20.1-Fabric-1.2.0.jar
✓ Downloaded 121 KB  StylishEffects-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 497 KB  FreshAnimations_v1.8.1.zip
✓ Downloaded 38.6 KB DisableCustomWorldsAdvice-4.1.jar
✓ Downloaded 79.1 KB VisualWorkbench-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 693 KB  Kiwi-1.20.1-fabric-11.1.1.jar
✓ Downloaded 397 KB  ForgeConfigAPIPort-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 73.3 KB BetterAdvancements-Fabric-1.20.1-0.3.2.161.jar
✓ Downloaded 84.9 KB DripSounds-1.19.4-0.3.2.jar
✓ Downloaded 852 KB  Patchouli-1.20.1-81-FABRIC.jar
✓ Downloaded 36.4 KB ash_api-fabric-3.0.2+1.20.1.jar
✓ Downloaded 1.15 MB YungsBetterDungeons-1.20-Fabric-4.0.1.jar
✓ Downloaded 118 KB  borderless-mining-1.1.8+1.20.1.jar
✓ Downloaded 460 KB  EasyShulkerBoxes-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 46.7 KB LeavesBeGone-v8.0.0-1.20.1-Fabric.jar
✓ Downloaded 395 KB  CITResewn-1.1.3+1.20.jar
✓ Downloaded 64.0 KB LegendaryTooltips-1.20.1-fabric-1.4.4.jar
✓ Downloaded 736 KB  completeconfig-2.5.0.jar
✓ Downloaded 1.15 MB YungsBetterDesertTemples-1.20-Fabric-3.0.1.jar
✓ Downloaded 383 KB  SlotCycler-v8.0.1-1.20.1-Fabric.jar
✓ Downloaded 1.54 MB YungsBetterOceanMonuments-1.20-Fabric-3.0.1.jar
✓ Downloaded 620 KB  NoChatReports-FABRIC-1.20.1-v2.2.2.jar
✓ Downloaded 179 KB  Statement-4.2.8+1.14.4-1.20.1.jar
✓ Downloaded 316 KB  VanillaTweaks-fabric-1.20.1-1.5.69.jar
✓ Downloaded 138 KB  connectiblechains-2.2.1+1.20.1.jar
✓ Downloaded 39.2 KB cyclepaintings-1.20.1-3.2.jar
✓ Downloaded 51.5 KB doubledoors-1.20.1-5.0.jar
✓ Downloaded 80.2 KB dynamic-fps-2.4.0.jar
✓ Downloaded 277 KB  customizableelytra-fabric-2.1.0+1.20.jar
No such file or directory (os error 2)
```

### Operating System

Linux

### Are you using the latest version of ferium?

- [X] I am using the latest version of ferium

### Additional Information

ferium 4.5.1

**Comment by theRookieCoder** [comment]
Thanks for reporting this, I'll look into it now. Also please do let me know of the other modpacks that have failed for you.

Modpacks can arbitrarily set their mod file paths to almost anything so it is difficult to make them all 100% compatible. Letting me know about incompatible modpacks helps me out a lot with improving support.

**Comment by insanity54** [comment]
Thank you. It was a couple weeks ago so I don't remember the other mods that failed but if I see it again I'll report it

**Comment by theRookieCoder** [comment]
Fixed in gorilla-devs/libium@fbe68a6

**Comment by theRookieCoder** [comment]
If you happen to remember the other modpacks (perhaps try your shell history) then please do check that the bug has been fixed in `ferium v4.5.2`

**Comment by karnzx** [comment]
not sure why 
i was try to install curseforge project 379757
install modpack in `/root/minecraft/modpacks` that currently empty
got error `No such file or directory (os error 2)`
but when install without option `Should overrides be installed?` which is `no`, it work.
```
root@ecs-6306:~/minecraft# ferium modpack add 379757
Checking modpack... ✓ (BlockFront - Historic Guns & Warfare)
Where should the modpack be installed to?
✔ Output Directory · /root/minecraft/modpacks
✔ Should overrides be installed? · yes
WARNING: Files in your output directory may be overwritten by modpack overrides
root@ecs-6306:~/minecraft# ferium modpack upgrade

Determining files to download... ✓ Fetched 1 mods
No such file or directory (os error 2)

root@ecs-6306:~/minecraft# ferium modpack rm
✔ Select which modpack to delete · CF 379757   BlockFront - Historic Guns & Warfare
root@ecs-6306:~/minecraft# ferium modpack add 379757
Checking modpack... ✓ (BlockFront - Historic Guns & Warfare)
Where should the modpack be installed to?
No such file or directory (os error 2)
root@ecs-6306:~/minecraft# ferium modpack add 379757
Checking modpack... ✓ (BlockFront - Historic Guns & Warfare)
Where should the modpack be installed to?
✔ Output Directory · /root/minecraft/modpacks
✔ Should overrides be installed? · no
root@ecs-6306:~/minecraft# ferium modpack upgrade

Determining files to download... ✓ Fetched 1 mods

Downloading 1 Mod Files

✓ Downloaded   111 MB  BlockFront-1.20.4-0.3.0.7a-RELEASE.jar

You can play this modpack using Minecraft 1.20.4 with neoforge-20.4.117-beta
root@ecs-6306:~/minecraft# ls modpacks/
mods  resourcepacks
```

**Comment by theRookieCoder** [comment]
Seems to be an issue with the modpack manifest. The modpack specifies that it has overrides in its `overrides` directory, but the zip file doesn't actually contain a directory called 'overrides' (or _anything_ at all apart from the manifest). It makes me wonder the modpack even exists if it just downloads and single mod and not even any resource packs or configs??

Anyways ferium shouldn't crash from something like this so I've implemented a fix. Let me know if any other modpacks fail.

**Comment by theRookieCoder** [comment]
Here's the fix 1045ccea47dbc944fbe058dd9e7c7b06c8160e9f

I won't dedicate a release to this, but if you need this fix immediately you can use the [CI build](https://github.com/gorilla-devs/ferium/actions/runs/8038169431).

**Comment by karnzx** [comment]
its work, thanks!
